WeatherForecastApp/Controller/LocationDetailViewController.swift

```diff
@@ -22,22 +24,27 @@ class LocationDetailViewController: UIViewController {
     @IBOutlet weak var summaryLabel: UILabel!
     @IBOutlet weak var imageView: UIImageView!
     @IBOutlet weak var pageControl: UIPageControl!
-    @IBOutlet weak var collectionView: UICollectionView!
     @IBOutlet weak var tableView: UITableView!
-    //var weatherLocation: WeatherLocation = WeatherLocation(name: "Current Location", latitude: 0.0, longitude: 0.0)
-    //var weatherLocations: [WeatherLocation] = []
-    var weatherDetail: WeatherDetail!
+
+    var weatherLocation: WeatherLocation!
```

Comment:
Never use force unwrapping always use optional wrapping. Force unwrapping is asserting that this variable or item will always have data and there is no possibility of this not having a value. Of which in this case if the user doesn't give permission to allow location capturing then weatherlocation will be nil. Meaning your app will crash because you told it that it will never happen that weatherLocation will be nil.